libs/langchain/langchain/callbacks/tracers/evaluation.py

```diff
@@ -133,9 +140,12 @@ def _persist_run(self, run: Run) -> None:
         run_ = run.copy()
         run_.reference_example_id = self.example_id
         for evaluator in self.evaluators:
-            self.futures.add(
-                self.executor.submit(self._evaluate_in_project, run_, evaluator)
-            )
+            if self.executor is None:
+                self._evaluate_in_project(run_, evaluator)
+            else:
+                self.futures.add(
+                    self.executor.submit(self._evaluate_in_project, run_, evaluator)
```

Comment:
Should the results be written to something that's thread safe? looks liek results end up in logged_eval_results default dict w/ append i don't really have a sense of whether .append is atomic

Reply:
Oh interesting ya I'll check